src/linux.rs

```diff
@@ -197,7 +197,7 @@ pub fn list_afinet_netifas() -> Result<Vec<(String, IpAddr)>, Error> {
 /// Retrieves the name of a interface address
 unsafe fn get_ifa_name(ifa: *mut *mut ifaddrs) -> Result<String, Error> {
```

Comment:
This is not compiling neither in my Ubuntu 20 locally or in the GitHub Actions workflow.
But I have a suggestion for your code I was able to run locally so I wanted to share with you:

What about replacing the whole function like this?

```rust
/// Retrieves the name of a interface address
fn get_ifa_name(ifa: *mut *mut ifaddrs) -> Result<String, Error> {
    let str = unsafe { (*(*ifa)).ifa_name as *const libc::c_char };
    let len = unsafe { strlen(str) };
    let slice = unsafe { std::slice::from_raw_parts(str as *const u8, len) };

    match String::from_utf8(slice.to_vec()) {
        Ok(s) => Ok(s),
        Err(e) => Err(Error::StrategyError(format!(
            "Failed to retrieve interface name. The name is not a valid UTF-8 string. {}",
            e.to_string()
        ))),
    }
}
```

Would this compile in your Raspberry Pi? I'm concerned about tests on a device like that because as of today I'm only able to test on Ubuntu locally and the GitHub Actions.

Reply:
I'll port this over to my Pi and check it out later today! Thanks.

Reply:
Awesome! Let me know how it goes! Feel free to push it as part of your PR so we get this merged and released if it works out!

Reply:
Success! Worked without any issues, and I was able to drop it into my other application. Thanks for the suggestion!

Reply:
Wohoo! Let's have this released then @alexkrob! Do you mind squashing commits into one? So Im able to merge this as is without any changes?

Reply:
All squashed. Happy merging!